lib/param_parsing/environment_test.go

```diff
@@ -97,3 +101,83 @@ func TestGetParamsFromEnvironment_log_level_from_env(t *testing.T) {
 		t.Errorf("Determined log level is not matching. Got %q, expected %q", params.LogLevel, expected)
 	}
 }
+
+func TestNoColorEnvVar(t *testing.T) {
+	envVarName := "NO_COLOR"
+	_ = os.Setenv(envVarName, "true")
+	goCommandArgs := []string{"run", "../../main.go", "--dry-run", "1.10.5"}
+
+	t.Logf("Testing %q env var", envVarName)
+
+	out, err := exec.Command("go", goCommandArgs...).CombinedOutput()
+	if err != nil {
+		t.Fatalf("Unexpected failure: \"%v\", output: %q", err, string(out))
+	}
+
+	_ = os.Unsetenv(envVarName)
+
+	matched, err := regexp.MatchString(ansiCodesRegex, string(out))
+	if err != nil {
+		t.Fatalf("Unexpected failure: \"%v\", output: %q", err, string(out))
+	}
+
+	if matched {
+		t.Errorf("Expected no ANSI color codes in output, but found some: %q", string(out))
+	} else {
+		t.Log("Success: no ANSI color codes in output")
+	}
+}
+
+func TestForceColorEnvVar(t *testing.T) {
+	envVarName := "FORCE_COLOR"
+	if color.SupportColor() {
+		_ = os.Setenv(envVarName, "true")
+		goCommandArgs := []string{"run", "../../main.go", "--dry-run", "1.10.5"}
+
+		t.Logf("Testing %q env var", envVarName)
+
+		out, err := exec.Command("go", goCommandArgs...).CombinedOutput()
+		if err != nil {
+			t.Fatalf("Unexpected failure: \"%v\", output: %q", err, string(out))
+		}
+
+		_ = os.Unsetenv(envVarName)
+
+		matched, err := regexp.MatchString(ansiCodesRegex, string(out))
+		if err != nil {
+			t.Fatalf("Unexpected failure: \"%v\", output: %q", err, string(out))
+		}
+
+		if !matched {
+			t.Errorf("Expected ANSI color codes in output, but found none: %q", string(out))
+		} else {
+			t.Log("Success: found ANSI color codes in output")
+		}
+	} else {
+		t.Skipf("Skipping test for %q env var as terminal doesn't support colors", envVarName)
```

Comment:
Does this get skipped in github actions? How does it behave with the SupportColor?

Reply:
> Does this get skipped in github actions?

Yes
```
2025-06-29T01:10:56.6664831Z === RUN   TestForceColorEnvVar
2025-06-29T01:10:56.6665348Z     environment_test.go:157: Skipping test for "FORCE_COLOR" env var as terminal doesn't support colors
2025-06-29T01:10:56.6665779Z --- SKIP: TestForceColorEnvVar (0.00s)
```

> How does it behave with the SupportColor?

Not sure I got the question. Would you mind rewording maybe?
At a sneak peak, if underlying library detects that terminal doesn't support color (escape) sequences, color output is not enabled.